tests/integration/cmd_namespace_test.go

```diff
@@ -180,6 +180,13 @@ ComponentSettings:
 					return out
 				}, 10*time.Second, 1*time.Second).Should(ContainSubstring(commonVar.Project))
 			})
+			It(fmt.Sprintf("should successfully list all the %ss in JSON format", commandName), func() {
+				Eventually(func() bool {
+					out := helper.Cmd("odo", "list", commandName, "-o", "json").ShouldPass().Out()
+					helper.MatchAllInOutput(out, []string{commonVar.Project, "\"active\": true", "\"active\": false"})
```

Comment:
As the output is in JSON, it is possible (and preferable) to use the JSON related functions (helper.JsonPathContentIs, ...)

If it is too difficult using gomega operators, you can do the check into the function (without any gomega.Expect), and return true when the result is ok, and just expect a True in Eventually(...).Should()

Reply:
@feloy it seems like even with `helper.JsonPathContentContain`, test immediately exit instead of retrying (as we would expect with `Eventually`). For below diff, the test immediately exits with failure:

```diff
diff --git a/tests/integration/cmd_namespace_test.go b/tests/integration/cmd_namespace_test.go
index 3439c6aa1..bf8f7f4c8 100644
--- a/tests/integration/cmd_namespace_test.go
+++ b/tests/integration/cmd_namespace_test.go
@@ -176 +176 @@ ComponentSettings:
-               Describe("list "+commandName, func() {
+               FDescribe("list "+commandName, func() {
@@ -186 +186 @@ ComponentSettings:
-                                       helper.MatchAllInOutput(out, []string{commonVar.Project, "\"active\": true", "\"active\": false"})
+                                       helper.JsonPathContentContain(out, "namespaces", commonVar.Project)
```

Reply:
@dharmit You will need to get inspiration from the implementation of these functions, but without any Expect, as this is what makes it fail immediately

Reply:
So I managed to make progress by nesting `Eventually` like this:
```go
It(fmt.Sprintf("should successfully list all the %ss in JSON format", commandName), func() {
	Eventually(func() bool {
		Eventually(func() bool {
			return commonVar.CliRunner.HasNamespaceProject(commonVar.Project)
		}, 10*time.Second, 1*time.Second).Should(Equal(true))

		out := helper.Cmd("odo", "list", commandName, "-o", "json").ShouldPass().Out()
		helper.JsonPathContentContain(out, "namespaces", commonVar.Project)
		return true
	}, 10*time.Second, 1*time.Second).Should(Equal(true))
})
```
but it's still failing. I don't think the `Eventually` aspect is being respected, in that it just fails if it can't find the namespace. It doesn't seem to be trying a bunch of times till `10*time.Second` is reached. 😕

Reply:
> @dharmit You will need to get inspiration from the implementation of these functions, but without any Expect, as this is what makes it fail immediately

As said before, you cannot use Expect inside the Eventually function, or it will raise the error immediately. Here, you are still using it (inside `JsonPathContentContain`).

You will need to pass a Gomega-free function to Eventually, only returning true/false, and Expect'ing on this true/false value

Reply:
> If it is too difficult using gomega operators, you can do the check into the function (without any gomega.Expect), and return true when the result is ok, and just expect a True in Eventually(...).Should()

Wish I had read your above command properly. @kadel helped me understand it better. 😞 
Fixed the test issue (at least locally.) 😕